data_steward/cdr_cleaner/cleaning_rules/drop_duplicate_ppi_questions_and_answers.py

```diff
@@ -0,0 +1,227 @@
+
+Original Issues: DC-539
+
+Intent is to drop the duplicated concepts for questions and answers created because of casing.
+"""
+
+# Python imports
+import logging
+
+# Third party imports
+from jinja2 import Environment
+
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.bq_utils import WRITE_TRUNCATE
+
+LOGGER = logging.getLogger(__name__)
+
+OBSERVATION = 'observation'
+ppi_modules = ['ppi_answers', 'ppi_questions']
+
+jinja_env = Environment(
+    # help protect against cross-site scripting vulnerabilities
+    autoescape=True,
+    # block tags on their own lines
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
+
+# Query to create tables in sandbox with the rows that will be removed per cleaning rule
+SANDBOX_PPI_ANSWERS = jinja_env.from_string("""
+CREATE OR REPLACE TABLE `{{project}}.{{sandbox_dataset}}.{{ans_table}}` as(
+SELECT value_source_concept_id, concept_id as new_value_source_concept_id, concept_id_2 as new_value_as_concept_id
+FROM `{{project}}.{{dataset}}.concept` c
+right join
+(SELECT value_source_concept_id, value_source_value, count( *)
+FROM `{{project}}.{{dataset}}.{{clinical_table_name}}`
+ join `{{project}}.{{dataset}}.concept` on (concept_id=value_source_concept_id)
+WHERE  invalid_reason is not null and questionnaire_response_id is not null
+group by 1,2
+) on (trim(lower(value_source_value))=lower(concept_code))
+join `{{project}}.{{dataset}}.concept_relationship` on (concept_id_1=concept_id)
+where vocabulary_id='PPI' and c.invalid_reason is null
+and relationship_id = 'Maps to value')
+""")
+
+SANDBOX_PPI_QUESTIONS = jinja_env.from_string("""
+CREATE OR REPLACE TABLE `{{project}}.{{sandbox_dataset}}.{{ques_table}}` as(
+SELECT observation_source_concept_id, concept_id as new_observation_source_concept_id,
+ concept_id_2 as new_observation_concept_id
+FROM `{{project}}.{{dataset}}.concept` c
+right join
+(SELECT observation_source_concept_id, observation_source_value, count( *)
+FROM `{{project}}.{{dataset}}.{{clinical_table_name}}`
+ join `{{project}}.{{dataset}}.concept` on (concept_id=observation_source_concept_id)
+WHERE  invalid_reason is not null and questionnaire_response_id is not null
+group by 1,2
+) on (trim(lower(observation_source_value))=lower(concept_code))
+join `{{project}}.{{dataset}}.concept_relationship` on (concept_id_1=concept_id)
+where vocabulary_id='PPI' and c.invalid_reason is null
+and relationship_id = 'Maps to')
+""")
+
+DELETE_DUPLICATE_ANSWERS = jinja_env.from_string("""
+select * 
+from `{{project}}.{{dataset}}.{{clinical_table_name}}` o
+where value_source_concept_id in (select value_source_concept_id 
```

Comment:
Since this isn't a DML statement, and you are truncating the query return to overwrite the table, shouldn't this be `NOT IN`?

Reply:
This should be, I am updating it